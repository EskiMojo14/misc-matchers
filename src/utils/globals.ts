// to avoid having separate jest and vitest suites, we create common definitions
// then mock during setup with the appropriate globals
import type {
  describe as jestDescribe,
  it as jestIt,
  expect as jestExpect,
  jest,
} from "@jest/globals";
import type {
  describe as vitestDescribe,
  it as vitestIt,
  expect as vitestExpect,
  vi,
  Mock as VitestMock,
} from "vitest";
import type { Satisfies } from "./types.typetest";

interface Each {
  <T extends Array<any> | [any]>(
    cases: ReadonlyArray<T>,
  ): (name: string, fn: (...args: T) => void) => void;
  <T>(cases: ReadonlyArray<T>): (name: string, fn: (value: T) => void) => void;
}
export declare const describe: (typeof vitestDescribe | typeof jestDescribe) & {
  each: Each;
};

// eslint-disable-next-line @typescript-eslint/no-unused-vars
type TestEach = Satisfies<
  typeof jestDescribe.each | typeof vitestDescribe.each,
  Each
>;

export declare const expect: typeof vitestExpect | typeof jestExpect;
export declare const it: typeof vitestIt | typeof jestIt;

export type Mock<R = any, A extends Array<any> = Array<any>, T = any> =
  | jest.Mock<(this: T, ...args: A) => R>
  | VitestMock<A, R>;

export declare const fn: {
  (): Mock;
  <R, A extends Array<any>, T>(fn?: (this: T, ...args: A) => R): Mock<R, A, T>;
};

// eslint-disable-next-line @typescript-eslint/no-unused-vars
type TestFn = Satisfies<typeof jest.fn | typeof vi.fn, typeof fn>;

throw new Error("This file should be mocked");
