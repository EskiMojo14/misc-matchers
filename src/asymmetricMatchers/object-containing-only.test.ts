import { describe, it, expect } from "@jest/globals";

describe("objectContainingOnly", () => {
  it("should check if object contains only the expected properties", () => {
    expect({ a: 1 }).toEqual(expect.objectContainingOnly({ a: 1, b: 2 }));

    expect(() => {
      expect({ a: 1, b: 2 }).toEqual(expect.objectContainingOnly({ a: 1 }));
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      expect({ a: 1, b: 2 }).toEqual(
        expect.objectContainingOnly({ a: 1, b: 3 }),
      );
    }).toThrowErrorMatchingSnapshot();

    expect({ a: 1, b: 2 }).toEqual(expect.not.objectContainingOnly({ a: 1 }));

    expect(() => {
      expect({ a: 1, b: 2 }).toEqual(
        expect.not.objectContainingOnly({ a: 1, b: 2 }),
      );
    }).toThrowErrorMatchingSnapshot();
  });
  it("should throw an error if expected value is not an object", () => {
    expect(() => {
      expect({}).toEqual(expect.objectContainingOnly(1));
    }).toThrowErrorMatchingSnapshot();
  });
  it("should throw an error if received value is not an object", () => {
    expect(() => {
      expect(1).toEqual(expect.objectContainingOnly({}));
    }).toThrowErrorMatchingSnapshot();
  });
});
