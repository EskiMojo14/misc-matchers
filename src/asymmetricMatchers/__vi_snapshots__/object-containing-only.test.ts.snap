// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`objectContainingOnly > should check if object contains only the expected properties 1`] = `expected { a: 1, b: 2 } to deeply equal objectContainingOnly<[object Object]>`;

exports[`objectContainingOnly > should check if object contains only the expected properties 2`] = `expected { a: 1, b: 2 } to deeply equal objectContainingOnly<[object Object]>`;

exports[`objectContainingOnly > should check if object contains only the expected properties 3`] = `expected { a: 1, b: 2 } to deeply equal not.objectContainingOnly{â€¦}`;

exports[`objectContainingOnly > should throw an error if expected value is not an object 1`] = `
<d>expect(</b><r>received</g><d>).</b>objectContainingOnly<d>(</b><g>1</g><d>)</b>

<b>Matcher error</b>: <g>Expected</g> value must be an object

Expected has type:  number
Expected has value: 1
`;

exports[`objectContainingOnly > should throw an error if received value is not an object 1`] = `expected 1 to deeply equal objectContainingOnly<[object Object]>`;
